lib/Parse/ParseDecl.cpp

```diff
@@ -1312,15 +1315,14 @@ bool Parser::parseTypeAttribute(TypeAttributes &Attributes, bool justChecking) {
       }
     }
     
-    // Recover by eating @foo when foo is not known.
+    // Recover by eating @foo(...) when foo is not known.
     consumeToken();
-      
-    // Recovery by eating "@foo=bar" if present.
-    if (consumeIf(tok::equal)) {
-      if (Tok.is(tok::identifier) ||
-          Tok.is(tok::integer_literal) ||
-          Tok.is(tok::floating_literal))
-        consumeToken();
+    if(Tok.is(tok::l_paren)) {
+      ParserPosition LParenPosition = getParserPosition();
+      skipSingle();
+      // If we found '->' after paren, it's likely a parameter of function type.
+      if (Tok.is(tok::arrow))
```

Comment:
Thanks! Added `getEndOfPreviousLoc() == Tok.getLoc()` condition, and `throws` after paren check.
How do you think now?